wormhole-connect/src/routes/operator.ts

```diff
@@ -221,6 +225,114 @@ export default class RouteOperator {
   }
 }
 
+// This caches successful quote results from SDK routes and handles multiple concurrent
+// async functions asking for the same quote gracefully.
+//
+// If we are already fetching a quote and a second hook requests the same quote elsewhere,
+// we queue up a Promise in `QuoteCacheEntry.pending` that we resolve when the original
+// quote request is resolved. This just prevents us from making redundant API calls when
+// multiple components or hooks are interested in a quote.
+class QuoteCache {
+  ttl: number;
+  cache: Record<string, QuoteCacheEntry>;
+  pending: Record<string, QuotePromiseHandlers[]>;
+
+  constructor(ttl: number) {
+    this.ttl = ttl;
+    this.cache = {};
+    this.pending = {};
+  }
+
+  quoteParamsKey(routeName: string, params: QuoteParams): string {
+    return `${routeName}:${params.sourceChain}:${params.sourceToken}:${params.destChain}:${params.destToken}:${params.amount}`;
```

Comment:
Do we need to include the nativeGas parameter in here if it exists? IIRC we fetch a quote whenever the nativeGas parameter changes

Reply:
good point. I'll add that.

Reply:
https://github.com/wormhole-foundation/wormhole-connect/pull/2535/commits/12d4d09bc2e2d73e790fce6afc2c613351764438